flexget/components/notify/notifiers/slack.py

```diff
@@ -95,25 +152,176 @@ class SlackNotifier:
+                        'text': {
+                            'type': 'mrkdwn',
+                            'text': block['section']['text'],
+                        },
+                    }
+
+                    if block['section'].get('image', {}).get('url'):
+                        section['accessory'] = {
+                            'type': 'image',
+                            'image_url': block['section']['image']['url'],
+                            'alt_text': block['section']['image']['alt_text'],
+                        }
+
+                    if block['section'].get('fields'):
+                        section['fields'] = []
+
+                        for field in block['section'].get('fields'):
+                            section['fields'].append({'type': 'mrkdwn', 'text': field})
+
+                    notification['blocks'].append(section)
+
+                elif block.get('image'):
+                    logger.trace('image block')
+                    image = {
+                        'type': 'image',
+                        'image_url': block['image']['url'],
+                        'alt_text': block['image']['alt_text'],
+                    }
+
+                    if block['image'].get('title'):
+                        image['title'] = {
+                            'type': 'plain_text',
+                            'text': block['image']['title'],
+                            'emoji': True,
+                        }
+
+                    notification['blocks'].append(image)
+
+                elif block.get('context'):
+                    logger.trace('context block')
+                    context = {'type': 'context', 'elements': []}
+
+                    for block_context in block.get('context'):
+                        if block_context.get('text'):
+                            context['elements'].append({
+                                'type': 'mrkdwn',
+                                'text': block_context['text'],
+                            })
+
+                        elif block_context.get('image'):
+                            context['elements'].append({
+                                'type': 'image',
+                                'image_url': block_context['image']['url'],
+                                'alt_text': block_context['image']['alt_text'],
+                            })
+
+                    notification['blocks'].append(context)
+
+                elif block.get('divider'):
+                    logger.trace('divider block')
+                    notification['blocks'].append({'type': 'divider'})
+
+        # version 1 is the Legacy formatting
+        else:
+            # username, channel, icon_emoji, and icon_url are deprecated and ignored by the Slack API,
+            # therefore they've been removed from the posted data
+            notification = {
+                'text': message,
+                'attachments': config.get('attachments'),
+            }
```

Comment:
In [https://api.slack.com/legacy/custom-integrations/messaging/webhooks](https://api.slack.com/legacy/custom-integrations/messaging/webhooks), it only says “Legacy custom integrations lack newer features and will be deprecated and possibly removed in the future.” It does not say “they are ignored by the Slack API.”